File: Programmers/기능개발.js
function solution(progresses, speeds) {
    let answer = [];


    while(speeds.length > 0) {
        for(let i=0;i<speeds.length;i++) {
            if(progresses[i] < 100) {
                progresses[i] += speeds[i];
            }
        }
        let deploy_count = 0;
        while(progresses[0] >= 100) {
            progresses.shift();
            speeds.shift();
            deploy_count++;
        }
        if(deploy_count > 0) {
            answer.push(deploy_count);
        }
    }
    return answer;
}

File: Programmers/기능개발.kt
import  java.util.*

class Programmers_stack {

    class Solution {
        fun solution(progresses: IntArray, speeds: IntArray): IntArray {
            val days = progresses.zip(speeds).map { (p, s) ->
                val left = 100 - p
                val div = left / s
                val mod = if (left % s > 0) 1 else 0
                div + mod
            }
            val stack = Stack<Pair<Int, Int>>()
            stack.add(Pair(days.first(), 1))

            for (day in days.drop(1)) {
                if (day <= stack.peek().first) {
                    val (a, b) = stack[stack.size-1]
                    stack[stack.size-1] = Pair(a, b+1)
                }
                else {
                    stack.add(Pair(day, 1))
                }
            }
            return stack.map { it.second }.toIntArray()
        }
    }

}